docs/DeveloperGuide.md

```diff
@@ -277,151 +277,202 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-      Use case resumes from step 4.
+* 2a. The list is empty.
+  \
+  Use case ends.
 
+* 4a. The given index is invalid.
+    * a1. MassLinkers prompts for another index.
+    * Step a1 is repeated until a valid index is given. \
+      Use case resumes from step 3.
 
-* a. At any time, Student chooses to close MassLinkers.
-    * a1. MassLinkers updates the written storage file.
-    * a2. MassLinkers closes the program. \
+* *a. At any time, Student chooses to close MassLinkers.
+    * a1. MassLinkers updates the storage file.
+      \
       Use case ends.
 
-**Use case 2: Delete a classmate**
+**Use case 3: Search batchmates via tags**
 
 **MSS**
 
-1. Student requests to list classmates.
-2. MassLinkers shows a list of classmates.
-3. Student requests to delete a specific classmate in the list.
-4. MassLinkers deletes the classmate using his index.
+1. Student requests the list of batchmates with tags.
+2. MassLinkers returns the list of batchmates associated with the particular tags.
    \
    Use case ends.
 
 **Extensions**
+* 1a. MassLinkers detects an invalid input. (No tags specified by the student)
+* 1a1. MassLinkers requests for at least one tag to be specified.
+* 1a2. Student inputs new tags.
+* Steps 1a1-1a2 are repeated until input is valid.
+  \
+  Use case resumes from step 2.
+* *a. At any time, Student chooses to close MassLinkers.
+    * a1. MassLinkers updates the storage file.
+      \
+      Use case ends.
+    
+**Use case 4: Add a tag to a batchmate**
 
-* 2a. The list is empty.
-\
-Use case ends.
+**MSS**
 
+1. Student chooses to add a tag to a batchmate.
+2. MassLinkers searches through the list of batchmates for the selected batchmate.
+3. MassLinkers adds tag to the batchmate.
+   \
+   Use case ends.
 
-* 3a. The given index is invalid.
-    * 3a1. MassLinkers shows an error message.
+**Extensions**
+* 2a. The list of batchmates is empty.
+* a1. MassLinkers warns adding a tag is invalid.
+  \
+  Use case ends.
+* 2b. MassLinkers is unable to find the batchmate.
+    * b1. MassLinkers prompts for new batchmate details.
+    * Step b1 is repeated until a batchmate is found.
       \
-      Use case resumes at step 2.
+      Use case resumes from step 2.
+
+* *a. At any time, Student chooses to close MassLinkers.
+    * a1. MassLinkers updates the storage file.
+      \
+      Use case ends.
+    
+**Use case 5: Mark a module completed**
+
+**MSS**
 
+1. Student requests the list of modules for a batchmate.
+2. Student requests to mark a module complete.
+3. MassLinkers marks the module completed.
+   \
+   Use case ends.
 
-* a. At any time, Student chooses to close MassLinkers.
-    * a1. MassLinkers updates the written storage file.
-    * a2. MassLinkers closes the program.
+**Extensions**
+* 2a. The list of modules is empty.
+    * a1. MassLinkers warns marking a module is invalid.
       \
       Use case ends.
+* 2b. The module selected is out of range.
```

Comment:
Should not include Ui details in use cases 🤔 ?

Reply:
Edited.